app/common/data/interfaces/user.py

```diff
@@ -169,19 +169,26 @@ def remove_grant_team_role_from_user(user: User, grant_id: uuid.UUID) -> None:
     db.session.expire(user)
 
 
-def remove_all_roles_from_user(user: User) -> None:
-    statement = delete(UserRole).where(UserRole.user_id == user.id)
-    db.session.execute(statement)
-    db.session.flush()
-    db.session.expire(user)
-
-
 def create_invitation(
     email: str,
-    organisation: Organisation | None = None,
     grant: Grant | None = None,
+    organisation: Organisation | None = None,
     role: RoleEnum | None = None,
 ) -> Invitation:
+    # Expire any existing invitations for the same email, organisation, and grant,
+    # filtering on NULL if org/grant not passed
+    stmt = update(Invitation).where(
+        and_(
+            Invitation.email == email,
+            Invitation.is_usable.is_(True),
+            Invitation.grant_id == grant.id if grant else Invitation.grant_id.is_(None),
+            Invitation.organisation_id == organisation.id if organisation else Invitation.organisation_id.is_(None),
```

Comment:
```suggestion
            (Invitation.grant_id == grant.id) if grant else Invitation.grant_id.is_(None),
            (Invitation.organisation_id == organisation.id) if organisation else Invitation.organisation_id.is_(None),
```

nit: a couple parens here might help make the order of operations a 🤏  cleaner. Genuinely unsure which way this falls right now - `(Invitation.grant_id == grant.id) if grant else Invitation.grant_id.is_(None)` or `Invitation.grant_id == (grant.id if grant else Invitation.grant_id.is_(None))`

Reply:
Ha fair point! I think for me your first suggestion is more readable to me personally and having just tried both the second one seems to lead to a bunch of SQLAlchemy and psycopg errors, so I'll go with the first one

Reply:
Haha gotcha, I misunderstood what you were saying 😆  yeah it looks like it's doing the first one (which is what we want), I've added the parens in to make it clearer 👍